server/src/weaverbird/backends/sql_translator/steps/duration.py

```diff
@@ -40,11 +40,20 @@ def translate_duration(
         columns_filter=[step.new_column_name]
     )
 
+    if sql_dialect == 'postgres':
+        interval_date = (
+            f" EXTRACT({step.duration_in} FROM ({step.end_date_column}-{step.start_date_column}))"
+        )
+    else:
+        interval_date = (
+            f" DATEDIFF({step.duration_in},"
+            f" to_timestamp({step.start_date_column}), to_timestamp({step.end_date_column}))"
+        )
```

Comment:
Last one : Postgres doesn't have a DATEDIFF & `to_timestamp` produce different result than the Snowflake equivalent.
However, Snowflake does support subtracting dates with the `-` operator & have an `EXTRACT` method.

So if the `to_timestamp` is unnecessary (why it's there ? aren't we supposed to use the duration step on date columns anyway ?), we could use the same SQL for both target :upside_down_face:

Reply:
> However, Snowflake does support subtracting dates with the - operator & have an EXTRACT method.
Source ? 
I've got
![image](https://user-images.githubusercontent.com/71276735/149352045-efeeaa71-ed52-4dac-aa4a-b98ac0d45044.png)

Reply:
I'll do some more tests, but we might have to keep it as is.

Reply:
add a comment : snowflake docs is wrong -> Cannot use '-' between timestamps.